### PR TITLE
fix(eslint): extend rules

### PR DESCRIPTION
**Pull request by korgara**
## Description
We need all those rules for Projects and Library MFEs.

## Rules added
https://eslint.org/docs/rules/array-callback-return
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-uses-react.md
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-uses-vars.md
## Checklist

- [x] Updated the version number in the `package.json`
- [x] Test any new configuration on a repo using [`npm link`](https://docs.npmjs.com/cli/link)

### CODE REVIEW

**Comment by benoitdeziel** [comment]
@korgara Can you provide a quick example in your description for the rules applied? It would be easier for us to review.

**Comment by benoitdeziel** [comment]
@davidmartinezmarin maybe we could provide a better template for the PR

**Comment by korgara** [comment]
@benoitdeziel those were added to local configs for Projects and Library mfes
Added rules description to the PRs description

File: eslint-config-landr/index.js [hunk 1]
```diff
@@ -100,6 +101,13 @@ module.exports = {
                 '@typescript-eslint/no-inferrable-types': 'error',
                 '@typescript-eslint/type-annotation-spacing': 'error',
                 '@typescript-eslint/no-use-before-define': 'error',
+                '@typescript-eslint/no-unused-vars': [
```
Comment (benoitdeziel) [inline eslint-config-landr/index.js:104]: based on the docs, you need to disable the base rule first
`"no-unused-vars": "off",`

Comment (korgara) [reply eslint-config-landr/index.js:106]: Thanks, good catch!
fixed

**Comment by korgara** [comment]
@MPeloquin added, good point👍

**Pull request merged**